upup/pkg/fi/cloudup/awstasks/iamrolepolicy.go

```diff
@@ -115,7 +114,19 @@ func (e *IAMRolePolicy) Find(c *fi.Context) (*IAMRolePolicy, error) {
 		if err != nil {
 			return nil, fmt.Errorf("error parsing PolicyDocument for IAMRolePolicy %q: %v", aws.StringValue(e.Name), err)
 		}
-		actual.PolicyDocument = fi.WrapResource(fi.NewStringResource(policy))
+
+		// Reformat the PolicyDocument by unmarshaling and re-marshaling to JSON.
```

Comment:
why is this necessary exactly?

Reply:
Good point. Will update.

Reply:
The document is just a JSON that is stored in AWS. In Direct and TF case, what you send is what you get back.
In the case of CloudFormation.... you get back a nice "optimised" JSON, that is no longer matching what was sent. Obviously, `kops update cluster` sees this as a change and tries to fix it.